xtask/src/publish.rs

```diff
@@ -58,74 +59,61 @@ fn remote_version(crate_name: &str) -> Option<String> {
     }
 }
 
-// Run cargo publish
-fn cargo_publish(params: &[&str]) {
-    // Run cargo publish
-    let mut cargo_publish = Command::new("cargo")
-        .arg("publish")
-        .arg("--color=always")
-        .args(params)
-        .stdout(Stdio::inherit()) // Send stdout directly to terminal
-        .stderr(Stdio::inherit()) // Send stderr directly to terminal
-        .spawn()
-        .expect("Failed to run cargo publish");
-
-    // Wait for cargo publish command to finish
-    let status = cargo_publish
-        .wait()
-        .expect("Failed to wait for cargo publish child process");
-
-    // If exit status is not a success, terminate the process with an error
-    if !status.success() {
-        // Use the exit code associated to a command to terminate the process,
-        // if any exit code had been found, use the default value 1
-        std::process::exit(status.code().unwrap_or(1));
-    }
-}
-
-// Publishes a crate
 fn publish(crate_name: String) {
-    // Run cargo publish --dry-run
-    cargo_publish(&["-p", &crate_name, "--dry-run"]);
+    // Perform dry-run to ensure everything is good for publishing
+    let dry_run_params = Params::from(["publish", "-p", &crate_name, "--dry-run"]);
+
+    run_cargo(
+        "publish",
+        dry_run_params,
+        HashMap::new(),
+        "Failed to run cargo publish --dry-run",
```

Comment:
This refactor is a good occasion to update the update message of this file, see the guidelines here https://github.com/tracel-ai/burn/blob/main/CONTRIBUTING.md#writing-expect-messages

Reply:
Got it! I just changed them, let me know if you think they're okay 👍